server-common/src/main/java/org/apache/kafka/server/common/StreamsGroupVersion.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.server.common;
+
+import java.util.Map;
+
+public enum StreamsGroupVersion implements FeatureVersion {
+
+    // Version 0 does disable "streams" groups (KIP-1071).
+    SGV_0(0, MetadataVersion.MINIMUM_VERSION, Map.of()),
+
+    // Version 1 enables "streams" groups (KIP-1071).
+    SGV_1(1, MetadataVersion.IBP_4_1_IV2, Map.of());
+
+    public static final String FEATURE_NAME = "streams.group.version";
```

Comment:
I would use `streams.version` to be consistent with the other ones.

Reply:
Seems to align to https://github.com/apache/kafka/pull/19509#discussion_r2050092192

Let's hear from Lucas about it.

Reply:
Silly question, but couldn't we just add this to group version? Do we need a whole new feature version for this? (Additionally, can we even set this if group version is not 1)